pkg/xds/sync/dataplane_sync_tracker_test.go

```diff
@@ -128,6 +130,58 @@ var _ = Describe("Sync", func() {
 
 			close(done)
 		}, 5)
+
+		It("should start only one watchdog per dataplane", func() {
+			// setup
+			var activeWatchdogs int32
+			tracker := NewDataplaneSyncTracker(func(dataplaneId core_model.ResourceKey, streamId int64) util_watchdog.Watchdog {
+				return WatchdogFunc(func(stop <-chan struct{}) {
+					atomic.AddInt32(&activeWatchdogs, 1)
+					<-stop
+					atomic.AddInt32(&activeWatchdogs, -1)
+				})
+			})
+
+			// when one stream for backend-01 is connected and request is sent
+			streamID := int64(1)
+			err := tracker.OnStreamOpen(context.Background(), streamID, "")
+			Expect(err).ToNot(HaveOccurred())
+			err = tracker.OnStreamRequest(streamID, &envoy.DiscoveryRequest{
+				Node: &envoy_core.Node{
+					Id: "default.backend-01",
+				},
+			})
+			Expect(err).ToNot(HaveOccurred())
+
+			// and when new stream from backend-01 is connected  and request is sent
+			streamID = 2
+			err = tracker.OnStreamOpen(context.Background(), streamID, "")
+			Expect(err).ToNot(HaveOccurred())
+			err = tracker.OnStreamRequest(streamID, &envoy.DiscoveryRequest{
+				Node: &envoy_core.Node{
+					Id: "default.backend-01",
+				},
+			})
+			Expect(err).ToNot(HaveOccurred())
+
+			// then only one watchdog is active
+			time.Sleep(50 * time.Millisecond) // wait for goroutines
```

Comment:
Is there a better way to syncronize here? Can this create flaky tests when running on slow instances in the CI?

Reply:
I tried to do it with channels, but the `// then watchdog is still active because other stream is opened` is hard to do it.

I don't think that it would cause problems, but I changed the simple wait to gomega `Eventually()` so we will be checking conditions for 5s every 10ms so even slower machines can check it.